Source/Expressive/Functions/String/ConcatFunction.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections;
+using System.Text;
+using Expressive.Expressions;
+
+namespace Expressive.Functions.String
+{
+    internal class ConcatFunction : FunctionBase
+    {
+        #region FunctionBase Members
+
+        public override string Name
+        {
+            get
+            {
+                return "Concat";
+            }
+        }
+
+        public override object Evaluate(IExpression[] parameters, Context context)
+        {
+            //We can accept 1 parameter that is an array
+            this.ValidateParameterCount(parameters, -1, 1);
+
+            StringBuilder sb = new StringBuilder();
+
+            //Recursive method if some parameters are arrays
+            Evaluate(sb, parameters, context);
+
+            //return the final full string
+            return sb.ToString();
+        }
+
+        protected virtual void Evaluate(StringBuilder sb, IEnumerable parameters, Context context)
+        {
+            foreach (var p in parameters)
+            {
+                object value;
+
+                if (p is IExpression)
+                {
+                    value = (p as IExpression).Evaluate(Variables);

```

Comment:
You could use pattern matching here and simply write:

```csharp
if (p is IExpression expression)
{
    value = expression.Evaluate(Variables);
```

It would be slightly more efficient but this is not something you need to change. Just pointing a feature in case you weren't aware of it 😄